### PR TITLE
Updated docs and sync

### PR DESCRIPTION
**Pull request by tikhomirovd**
1. Added doc setting for HypEx
2. Added new tutorials to doc 
3. Fixed warnings with "Ipythonx" while building doc
4. Fixed link to doc scipy (it was warning) 
5. Added example of using Matching and more info in docs
6. Added custom func for weight features for matching 
7. Added base filtration
8. Fixed info_col
9. Matching without replacement

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,13 +43,14 @@ numpy = [
...
 scikit-learn = ">=0.22"
-lightgbm = ">=2.3, <=3.2.1"
+lightgbm = ">=2.3"
 catboost = ">=0.26.1"
 optuna = "*"
 torch = [
     {platform = "win32", python = "3.6.1", version = "1.7.0"},
     {version = ">=1.0.0"}
 ]
+faiss-cpu="*"
```
Comment (dev-rinchin) [inline pyproject.toml:53]:
```suggestion
faiss-cpu={version = "*", optional = true}
```

Comment (dev-rinchin) [reply pyproject.toml:53]: 1) add section for hypex requirements:
```
hypex = [
    "faiss-cpu"
]
```
2) add "hypex" to requiremets for "all"

Comment (tikhomirovd) [reply pyproject.toml]: Done

File: pyproject.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ numpy = [
     {version = ">=1.22", python = ">=3.10"}
 ]
 scikit-learn = ">=0.22"
-lightgbm = ">=2.3, <=3.2.1"
+lightgbm = ">=2.3"
```
Comment (dev-rinchin) [inline pyproject.toml:46]: please leave it as it is:
```suggestion
lightgbm = ">=2.3, <=3.2.1"
```

Comment (tikhomirovd) [reply pyproject.toml]: Done

**Pull request merged**